src/snowflake/cli/api/project/definition_manager.py

```diff
@@ -25,18 +24,30 @@ class DefinitionManager:
     project_root: Path
     _project_config_paths: List[Path]
 
-    def __init__(self, project_arg: Optional[str] = None) -> None:
-        search_path = os.getcwd()
-        if project_arg:
-            search_path = os.path.abspath(project_arg)
-        project_root = DefinitionManager.find_project_root(Path(search_path))
-        if not project_root:
-            raise MissingConfiguration(
-                f"Cannot find project definition (snowflake.yml). Please provide a path to the project or run this command in a valid project directory."
-            )
+    def __init__(
+        self, project_root: Path | None = None, project_files: List[Path] | None = None
+    ) -> None:
+        """
+        :param project_root: project root where the project lives
+        :param project_files: List of project definition files, relative to project root
+        """
+        if project_root is None:
+            project_root = Path.cwd()
 
         self.project_root = project_root
-        self._project_config_paths = self._find_definition_files(self.project_root)
+
+        if project_files is None:
+            project_files = self._find_definition_files(self.project_root)
+        else:
+            project_files = [self.project_root / p for p in project_files]
```

Comment:
As a CLI user, this might be a bit confusing. When I am entering project file in the CLI, I use autocomplete to locate the file I would like to use as a project file. This would not work if I am NOT in the project root, and I am explicitly specifying --project in the command.  example: 
```
cd project-folder/src
snow app run --project ../ --proj-file ../config/project.yml
```
To add to this, we can see usecases where you are managing multiple snow CLI projects, but you also have a set of yml files for different situations/environments in a separate folder that apply to multiple projects, and you want to switch between them. With the above approach, you cannot specify an absolute project file, specially for overrides.